### PR TITLE
add support for parsing services

### PR DESCRIPTION
**Pull request by dmccartney**
Hello @nicks, @dpup, 

Please review the following commits I made in branch 'daniel-add-services'.

91fd8faae25d0571a21dab94a38da9fa43c31b6d (2016-05-18 14:26:03 -0700)
add support for parsing services

R=@nicks
R=@dpup

### CODE REVIEW

File: lib/Project.js [hunk 1]
```diff
@@ -232,6 +232,13 @@ Project.prototype._discoverTypes = function (proto, package) {
     var eName = helper.joinPackage(package, e.getName())
     this._typesByName[eName] = e
   }, this)
+
+  if (!proto.getServices) {
```
Comment (nicks) [inline lib/Project.js]: this doesn't make sense to me? why would the method not exist?

File: lib/Project.js [hunk 2]
```diff
@@ -253,26 +260,54 @@ Project.prototype._resolveTypes = function (proto, package) {
...
+        ' on message ' + message.getName() +
+        ' : ' + field.getRawType())
+      field.setTypeDescriptor(type)
     }, this)
 
     this._resolveTypes(message, helper.joinPackage(package, message.getName()))
   }, this)
+
+  // resolve type references on service descriptors
+  if (!proto.getServices) {
```
Comment (nicks) [inline lib/Project.js]: ditto here

**Comment by nicks** [comment]
lg

File: lib/Project.js (hunk 1 above)
Comment (dmccartney) [inline lib/Project.js]: `proto` might be a `MessageDescriptor`, and `.getServices` is only present when `proto` is a `ProtoDescriptor`
```
@param {MessageDescriptor|ProtoDescriptor} proto
```

the function recurses on `message` (see `_discoveryTypes(..)` at L226) to resolve nested message types.  but we don't support nesting `service` definitions within `messages`, so there's no `.getServices` on the `MessageDescriptor`.

**Comment by dmccartney** [comment]
@nicks I pulled the tests into their own test file and bumped the version number

File: lib/Project.js (hunk 1 above)
Comment (dpup) [inline lib/Project.js]: instance of check would be more self documenting.

**Pull request merged**